File: backend/app/Models/ToDoComponent.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ToDoComponent extends Model{
    protected $table = 'todocomponents';

    /**
     * Set the autotimestamps to false
     * 
     * @var bool $timestamps
     */
    public $timestamps = false;

    /**
     * The fillable fields.
     * 
     * @var  array $fillable
     */
    protected $fillable = [
        'title', 'description', 'priority', 'image_url',
    ];

    public static function boot(){

        /* call parent boot method */
        parent::boot();

    }
}

File: frontend/lib/model/todocomponent.dart
class ToDoComponent {
  int? id;
  String title;
  String description;
  String priority;
  String imageUrl;

  ToDoComponent(
      {this.id,
      required this.title,
      required this.description,
      required this.priority,
      required this.imageUrl});

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'priority': priority,
        'image_url': imageUrl
      };

  static ToDoComponent fromSnap(Map<String, dynamic> json) {
    return ToDoComponent(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        priority: json['priority'],
        imageUrl: json['image_url']);
  }
}
